### PR TITLE
Configuring logic to create user or authenticate

### PR DESCRIPTION
**Pull request by sankassio99**
- Removing build files from versioning
- AuthenticateUser
- RegisterUserIfDoesExists
- Implementing TDD
- Decoupling from Cognito

### CODE REVIEW

File: src/SignIn/Repositories/CognitoSignInRepository.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Amazon.CognitoIdentityProvider;
+using Amazon.CognitoIdentityProvider.Model;
+using SignIn.Contracts;
+
+namespace SignIn.Repositories;
+
+public class CognitoSignInRepository : ISignInRepository
```
Comment (IcaroSousa) [inline src/SignIn/Repositories/CognitoSignInRepository.cs:10]: Hello!

As it consumes an external exchangeable service provider, it seems to be a service to me not a repository.

What do you think about renaming it  moving it to a service layer ?
But you know, it is just a naming detail, overall, this class is great ;)

Congrats!

**Review by IcaroSousa — COMMENTED** [review]
This PR is great man!

Congrats 🥂

File: src/SignIn/Repositories/CognitoSignInRepository.cs (hunk 1 above)
Comment (sankassio99) [reply src/SignIn/Repositories/CognitoSignInRepository.cs:10]: Hello @IcaroSousa !

I was with this doubt, the "repository" nomenclature is used more for persistence in the database, right?

I think is good to change this name, thanks for the comment bro!

Comment (IcaroSousa) [reply src/SignIn/Repositories/CognitoSignInRepository.cs:10]: Yep!

We can close this PR and change it in another PR later. It is not something critical at all 😊;

**Pull request merged**